FireboltNETSDK/Client/FireboltConnection.cs

```diff
@@ -96,18 +101,32 @@ public string Account
             set;
         }
 
+        internal bool IsSystem
+        {
+            get => _isSystem;
+        }
+
         public string? AccountId
         {
             get
             {
                 if (_accountId == null && Account != null && _isSystem)
                 {
-                    _accountId = Client?.GetAccountIdByNameAsync(Account, CancellationToken.None).GetAwaiter().GetResult().id;
+                    GetAccountIdByNameResponse account = Client.GetAccountIdByNameAsync(Account, CancellationToken.None).GetAwaiter().GetResult();
+                    _accountId = account.id;
+                    _infraVersion = account.infraVersion;
                 }
-                return _isSystem ? _accountId : null;
+                return _accountId;
+                //return _isSystem ? _accountId : null;
             }
+            set => _accountId = value;
         }
 
+        internal int InfraVersion
+        {
+            get => _infraVersion;
+            set => _infraVersion = value;
+        }
```

Comment:
No * 3. Special values is the way to hell. 
I'd prefer to use time machine, travel 3 years ago and guarantee that the infra version is supported by the server from the first day. But this is not the case. So, I assume in the beginning that the infra version is 1 and then update it if it is reported by the server. 

I however agree with you that this is very bad that setting infra version is done as a side effect of getting account ID. I wanted to improve it but decided to postpone the implementation to avoid unwanted side effects.

Reply:
You can use `private int? _infraVersion` without initialising it or whatever you like to make sure we'll never get this value from the server. But this getter relies on a specific state right now and it's not obvious.

Reply:
This will make things more complicated. I will have to validate whether int is defined or not. It is (almost) exactly as null in java. I want to avoid it. If InfraVersion is not returned from server it is version 1. If returned it is the version that is returned from server. This is easier and safer.

Reply:
Changed as you suggested but the initial value of version is set to 0.

Reply:
> Changed as you suggested

I suggested doing a request if InfraVersion hasn't been initialised when it's being accessed elsewhere. However, you at least made it not as dangerous as before so I'll resolve this conversation.